File: app/src/main/java/dev/ragnarok/fenrir/model/criteria/DocsCriteria.java
package dev.ragnarok.fenrir.model.criteria;

public class DocsCriteria extends Criteria {

    private final int accountId;

    private final int ownerId;

    private Integer filter;

    public DocsCriteria(int accountId, int ownerId) {
        this.accountId = accountId;
        this.ownerId = ownerId;
    }

    public Integer getFilter() {
        return filter;
    }

    public DocsCriteria setFilter(Integer filter) {
        this.filter = filter;
        return this;
    }

    public int getOwnerId() {
        return ownerId;
    }

    public int getAccountId() {
        return accountId;
    }
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/model/criteria/DocsCriteria.kt
package dev.ragnarok.fenrir.model.criteria

class DocsCriteria(val accountId: Int, val ownerId: Int) : Criteria() {
    var filter: Int? = null
        private set

    fun setFilter(filter: Int?): DocsCriteria {
        this.filter = filter
        return this
    }
}